SectionedCollectionView/Views/SectionedCollectionView.swift

```diff
@@ -168,133 +151,55 @@ class SectionedCollectionView: UIView {
-            .subscribe(onNext: { indexPath in
-                let selectedItemsCount = self.sections.value.compactMap({ sections -> [CustomData] in
-                    return sections.selectedItems().items
-                }).flatMap({ $0 }).count
-                
-                if (self.sections.value[indexPath.section].items[indexPath.row].selected || self.settings.data.selectedLimit == nil || selectedItemsCount < self.settings.data.selectedLimit!) {
-                    self.sections.value[indexPath.section].items[indexPath.row].selected = !self.sections.value[indexPath.section].items[indexPath.row].selected
-                } else {
-                    self._limitReached.asObserver().onNext(())
-                }
-            })
-            .disposed(by: disposeBag)
-    }
-    
-    // MARK: - Public override variables
-    
-    var headerReferenceSize: CGSize {
-        
-        set {
-            let collectionViewFlowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
-            collectionViewFlowLayout?.headerReferenceSize = newValue
-        }
-        
-        get {
-            let collectionViewFlowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
-            return collectionViewFlowLayout?.headerReferenceSize ?? .zero
+}
+
+extension SectionedCollectionView: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let selectedItemsCount = self.sections.compactMap({ sections -> [CustomData] in
+            return sections.selectedItems().items
+        }).flatMap({ $0 }).count
+        
+        if (self.sections[indexPath.section].items[indexPath.row].selected || self.settings.data.selectedLimit == nil || selectedItemsCount < self.settings.data.selectedLimit!) {
+            self.sections[indexPath.section].items[indexPath.row].selected = !self.sections[indexPath.section].items[indexPath.row].selected
+            self.delegate?.selectedItems(selected: self.sections.map({ section -> [CustomData] in
+                return section.selectedItems().items
+            }).flatMap({ $0 }))
+            collectionView.reloadData()
+        } else {
+            self.delegate?.limitReached()
         }
-        
     }
-    
-    var footerReferenceSize: CGSize {
-        
-        set {
-            let collectionViewFlowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
-            collectionViewFlowLayout?.footerReferenceSize = newValue
-        }
-        
-        get {
-            let collectionViewFlowLayout = collectionView.collectionViewLayout as? UICollectionViewFlowLayout
-            return collectionViewFlowLayout?.footerReferenceSize ?? .zero
-        }
-        
+}
+
+extension SectionedCollectionView: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
```

Comment:
Add an empty line here and one at the bottom of the extension if there isn't already